docker/migrations/1_deploy_712_lib_mock.js

```diff
@@ -0,0 +1,5 @@
+var Contract = artifacts.require("PaymentEip712LibMock");
```

Comment:
Just a warning. Truffle migration script would record the starting number of the file name, so 1 in this case. The next time it would start with the next number. This file would collide with the original `1_initial_checks_and_migrations.js` file with the number 1. I think it is fine since your test setup does not care about adding new migration and start from the middle step of migration scripts. The tests seems to always want to run all migrations at once. So just a warning on this.

(PS. so actually there is a line `await deployer.deploy(Migrations);` in `1_initial_checks_and_migrations.js` which would deploy the `Migrations` contract that records the number.)

Reply:
@boolafish okay that make sense! Okay so my understanding for the mock contract is:

* You only really need one migration to load them?(e.g I don't need to load the other framework contracts, the `require` handles that?)
* Using the above docker+volume mount method, I should be able to ignore the migration indexing situation. I'm assuming that any new mock contracts I'd want to deploy will be into its own volume and mount it back into its own `plasma-contracts` container to deploy. So example, for the payment signing, I just want the `PaymentEIP712LibMock` to test, and I assume in the future another mock to test something else(so new container and volume)

Alternatively, if there is alot of mock contracts we want to deploy against, I could change this to just do a normal set of migrations and ensure I deploy the `Migrations` contract and index all the mock migrations appropriately.

Reply:
> You only really need one migration to load them?(e.g I don't need to load the other framework contracts, the require handles that?)

I guess. But it really depends on what contract you'd like to test. I think most of the ones you're interested should be fine without other contracts existing.

> I'm assuming that any new mock contracts I'd want to deploy will be into its own volume and mount it back into its own plasma-contracts container to deploy.

Oh interesting, so your strategy is to have one container for each test? So payment singing has one container, and the other would spin up another?

But anyway, I agree the migration indexing is not really a concern of the test here. Just bring up a warning if anything weird happens, this can potentially be the cause.

Reply:
> Oh interesting, so your strategy is to have one container for each test? So payment singing has one container, and the other would spin up another?

It's the strategy for now as I'm not entirely sure how many mock contracts we will wnat to eventually deploy. If there is a substantial amount probably makes sense to follow the migration index mechanics and deploy the mock contracts as a whole set (one container for plasma framework contract testing, another for mock contract testing)